apps/frontend/src/app/2_molecules/AssetAmountInput/AssetAmountInput.tsx

```diff
@@ -46,6 +48,21 @@ export const AssetAmountInput: FC<AssetAmountInputProps> = ({
     [chainId],
   );
 
+  const handleMaxClick = useCallback(() => {
```

Comment:
What about having only one handler that sends either custom amount or max amount (when max button clicked) and then have the logic to validate if `maxAmount` was selected in the parent component? This would prevent having to send a prop callback to tell if button was clicked